content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+
+Bevy has always used [WGPU](https://github.com/gfx-rs/wgpu), a native GPU abstraction layer with support for most graphics backends: Vulkan, Metal, DX12, OpenGL, WebGL2, and WebGPU (and WIP DX11 support). But the old renderer hid it behind our own hardware abstraction layer. In practice, this was largely just a mirror of the WGPU api. It gave us the ability to build own our graphics backends without bothering the WGPU folks, but in practice it created a lot of pain (due to being an imperfect mirror), overhead (due to introducing a dynamic api and requiring global mutex locks over gpu resource collections), and complexity (bevy_render -> wgpu -> Vulkan). In return, we didn't get many practical benefits ... just slightly more autonomy.
+
+The truth of the matter is that WGPU already occupies _exactly_ the space we want it to:
+* Multiple backends, with the goal to support as many platforms as possible
+* A "baseline" feature set that works almost everywhere with a consistent api
+* A "limits" and "features" system that enables opting in to arbitrary (sometimes backend-specific features) and detecting when those features are available. This will be important when we start adding things like raytracing and VR support.
+* A modern GPU api, but without the pain and complexity of raw Vulkan. Perfect for user-facing Bevy renderer extensions.
+
+However initially there were a couple of reasons not to make it our "public facing api":
+* **Complexity**: WGPU used to be built on top of gfx-hal (on older GPU abstraction layer also built and managed by the WGPU team). These multiple layers of abstraction in multiple repos made contributing to and reasoning about the internals difficult.
+* **Licensing**: WGPU used to be licensed under the "copyleft" MPL license, which created concerns about integration with proprietary graphics apis (such as consoles like the Switch).
+* **WebGL2 Support**: WGPU used to not have a WebGL2 backend. Bevy's old renderer had a custom WebGL2 backend and we weren't willing to give the Web platform up.
+
+_Almost immediately_ after we voiced these concerns, @kvark kicked off a [relicensing effort](https://github.com/gfx-rs/wgpu/issues/392) that switched WGPU to the Rust-standard dual MIT/Apache-2.0 license. They also removed gfx-hal in favor of a [much simpler and flatter architecture](https://gfx-rs.github.io/2021/08/18/release-0.10.html). Soon after, @zicklag [added a WebGL2 backend](https://github.com/gfx-rs/wgpu/pull/1686). Having resolved all of my remaining hangups, it was clear to me that @kvark's priorities were aligned with mine and that I could trust them to adjust to feedback. And now that WGPU has a flatter architecture, I feel more comfortable forking and maintaining it under the Bevy umbrella if that ever becomes necessary. But I doubt we will ever need to ... WGPU has proven to be a responsive partner and I'm very comfortable letting @kvark own this area, given his expertise.
```

Comment:
Good call. I intentionally use gender-neutral pronouns wherever possible (unless specifically asked to do otherwise by individuals). I'll change this to "their".